pkg/backend/watch.go

```diff
@@ -108,8 +102,83 @@ func Watch(ctx context.Context, b Backend, stack Stack, op UpdateOperation,
 			display.PrintfWithWatchPrefix(time.Now(), "",
 				op.Opts.Display.Color.Colorize(colors.SpecImportant+"Update complete."+colors.Reset+"\n"))
 		}
-
 	}
 
 	return nil
 }
+
+func watchPaths(root string, paths []string) (chan string, func(), result.Result) {
+	args := []string{"--origin", root}
+	for _, p := range paths {
+
+		watchPath := ""
+		if path.IsAbs(p) {
+			watchPath = p
+		} else {
+			watchPath = path.Join(root, p)
+		}
+
+		args = append(args, "--watch", watchPath)
+	}
+
+	watchCmd, err := getWatchUtil()
+	if err != nil {
+		return nil, nil, result.FromError(err)
+	}
+
+	cmd := exec.Command(watchCmd, args...)
+	cmdutil.RegisterProcessGroup(cmd)
+	reader, _ := cmd.StdoutPipe()
+
+	scanner := bufio.NewScanner(reader)
+	events := make(chan string)
+	go stdoutToChannel(scanner, events)
+	err = cmd.Start()
+	if err != nil {
+		return nil, nil, result.FromError(fmt.Errorf("error starting pulumi-watch: %w", err))
+	}
+
+	stop := func() {
+		err := cmd.Process.Kill()
+		contract.AssertNoErrorf(err, "Unexpected error stopping pulumi-watch process: %v", err)
+	}
+
+	return events, stop, nil
+}
+
+const windowsGOOS = "windows"
+
+func getWatchUtil() (string, error) {
+	program := "pulumi-watch"
+	if runtime.GOOS == windowsGOOS {
+		program = "pulumi-watch.exe"
+	}
+
+	watchCmd, err := exec.LookPath("pulumi-watch")
+	if err == nil {
+		return watchCmd, nil
+	}
+
+	exePath, exeErr := os.Executable()
+	if exeErr == nil {
+		fullPath, fullErr := filepath.EvalSymlinks(exePath)
+		if fullErr == nil {
+			candidate := filepath.Join(filepath.Dir(fullPath), program)
+
+			// Let's see if the file is executable. On Windows, os.Stat() returns a mode of "-rw-rw-rw" so on
+			// on windows we just trust the fact that the .exe can actually be launched.
+			if stat, err := os.Stat(candidate); err == nil &&
```

Comment:
Should we give a better error here that the file was found but wasn't executable? Rather than just "Could not locate"

Reply:
Added that case:

```
fmt.Errorf("Could not locate an executable pulumi-watch, found %v without execute bit", fullPath)
```